flyteadmin/pkg/repositories/gormimpl/execution_repo.go

```diff
@@ -45,13 +54,21 @@ func (r *ExecutionRepo) Create(ctx context.Context, input models.Execution, exec
 func (r *ExecutionRepo) Get(ctx context.Context, input interfaces.Identifier) (models.Execution, error) {
 	var execution models.Execution
 	timer := r.metrics.GetDuration.Start()
+
+	isolationFilter := util.GetIsolationFilter(ctx, isolation.DomainTargetResourceScopeDepth, executionColumnNames)
+
 	tx := r.db.WithContext(ctx).Where(&models.Execution{
 		ExecutionKey: models.ExecutionKey{
 			Project: input.Project,
 			Domain:  input.Domain,
 			Name:    input.Name,
 		},
-	}).Take(&execution)
+	})
+	if isolationFilter != nil {
+		cleanSession := tx.Session(&gorm.Session{NewDB: true})
```

Comment:
Another key implementation detail is that a new session must be declared on the transaction in order to properly separate the newly injected where clauses within a paranthesis. If this is not done we've see gorm try and be smart and combine/deduplicate clauses which causes bugs.